File: module_base/src/main/java/com/app/base/router/AppRouterImpl.java
package com.app.base.router;

import android.net.Uri;

import com.alibaba.android.arouter.facade.template.IProvider;

/**
 * @author Ztiany
 * Email: 1169654504@qq.com
 * Date : 2017-11-04 13:45
 */
public class AppRouterImpl implements AppRouter {

    @Override
    public IPostcard build(String path) {
        return RouterManager.build(path);
    }

    @Override
    public IPostcard build(Uri path) {
        return RouterManager.build(path);
    }

    @Override
    public void inject(Object object) {
        RouterManager.inject(object);
    }

    @Override
    public <T extends IProvider> T navigation(Class<T> iProviderClass) {
        return RouterManager.navigation(iProviderClass);
    }

}

File: common_sdk/src/main/java/com/app/base/router/AppRouterImpl.kt
package com.app.base.router

import com.android.common.router.Navigator
import java.util.*

/**
 * @author Ztiany
 */
internal class AppRouterImpl : AppRouter {

    override fun <T : Navigator> get(clazz: Class<T>): T? {
        return ServiceLoader.load(clazz).firstOrNull()
    }

    override fun <T : Navigator> require(clazz: Class<T>): T {
        return ServiceLoader.load(clazz).firstOrNull() ?: throw NullPointerException("implementation of $clazz not found.")
    }

}